src/components/menu-bar/menu-bar.jsx

```diff
@@ -241,16 +242,43 @@ class MenuBar extends React.Component {
                 id="gui.menuBar.new"
             />
         );
+        const shareMessage = (
+            <FormattedMessage
+                defaultMessage="Share"
+                description="Label for project share button"
+                id="gui.menuBar.share"
+            />
+        );
+        const isSharedMessage = (
+            <FormattedMessage
+                defaultMessage="Shared"
+                description="Label for shared project"
+                id="gui.menuBar.isShared"
+            />
+        );
         const shareButton = (
             <Button
-                className={classNames(styles.shareButton)}
+                className={classNames(
+                    styles.menuBarButton,
+                    styles.shareButton,
+                    {[styles.shareButtonIsShared]: this.props.isShared}
+                )}
                 onClick={this.props.onShare}
             >
-                <FormattedMessage
-                    defaultMessage="Share"
-                    description="Label for project share button"
-                    id="gui.menuBar.share"
-                />
+                {this.props.isShared ? isSharedMessage : shareMessage}
```

Comment:
I think with jsx it is generally preferred to inline everything, it makes it easier to read. Could you change this to inline the two formatted messages?